### PR TITLE
fetcher: Dockerfile and deployment

### PR DESCRIPTION
**Pull request by apelisse**
Create Dockerfile to build container and deployment to set-up

---
This change is [Reviewable](https://reviewable.io/reviews/kubernetes/test-infra/447)

### CODE REVIEW

**Comment by spxtr** [comment]
I'd recommend using a makefile instead of deploy.sh.

---

Review status: 0 of 4 files reviewed at latest revision, 5 unresolved discussions.

---

*[github-stats/fetcher/deploy.sh, line 20 \[r1\]](https://reviewable.io:443/reviews/kubernetes/test-infra/447#-KPxa0SgcR0cejKSj8nc:-KPxa0SgcR0cejKSj8nd:b-994rms) ([raw file](https://github.com/kubernetes/test-infra/blob/5eb8377311aa0d8d1fc9ac56d0d3f4e8c2ff09eb/github-stats/fetcher/deploy.sh#L20)):*
> ```Shell
> set -o nounset
> 
> IMAGE="${1:-gcr.io/google_containers}/fetcher-$(date +%Y-%M-%d)-$(git rev-parse --verify --short HEAD)"
> ```

If you're allowing for an argument to the script then you should have a comment at the top saying as much.

---

*[github-stats/fetcher/deploy.sh, line 24 \[r1\]](https://reviewable.io:443/reviews/kubernetes/test-infra/447#-KPxaCCVtO0K6ItMVwH9:-KPxaCCVtO0K6ItMVwHA:b-col8ot) ([raw file](https://github.com/kubernetes/test-infra/blob/5eb8377311aa0d8d1fc9ac56d0d3f4e8c2ff09eb/github-stats/fetcher/deploy.sh#L24)):*
> ```Shell
> set -o xtrace
> 
> CGO_ENABLED=0 go build .
> ```

This is assuming you're running it from the same directory.

---

*[github-stats/fetcher/deployment.yaml, line 16 \[r1\]](https://reviewable.io:443/reviews/kubernetes/test-infra/447#-KPxarGfz-NhYxbHqYYd:-KPxarGfz-NhYxbHqYYe:bt3mxh3) ([raw file](https://github.com/kubernetes/test-infra/blob/5eb8377311aa0d8d1fc9ac56d0d3f4e8c2ff09eb/github-stats/fetcher/deployment.yaml#L16)):*
> ```YAML
>         command:
>         - /fetcher
>         - --token=$(GITHUB_TOKEN)
> ```

This is a bit scary. If anyone manages to see our command line flag then they see our token. Instead, you should mount the secret into a file and pass the location of that.

---

*[github-stats/fetcher/Dockerfile, line 15 \[r1\]](https://reviewable.io:443/reviews/kubernetes/test-infra/447#-KPx_gROqTPbx_mBjxTa:-KPx_gROqTPbx_mBjxTb:b3e8ck2) ([raw file](https://github.com/kubernetes/test-infra/blob/5eb8377311aa0d8d1fc9ac56d0d3f4e8c2ff09eb/github-stats/fetcher/Dockerfile#L15)):*
> ```Dockerfile
> # limitations under the License.
> 
> FROM google/debian:jessie
> ```

Why debian? Why not go with something lighter such as alpine?

---

*[github-stats/fetcher/Dockerfile, line 16 \[r1\]](https://reviewable.io:443/reviews/kubernetes/test-infra/447#-KPx_k0dVK01hQyOPD8t:-KPx_k0dVK01hQyOPD8u:b-m1zdzj) ([raw file](https://github.com/kubernetes/test-infra/blob/5eb8377311aa0d8d1fc9ac56d0d3f4e8c2ff09eb/github-stats/fetcher/Dockerfile#L16)):*
> ```Dockerfile
> 
> FROM google/debian:jessie
> RUN apt-get update
> ```

The apt-get update needs to be in the same command as the apt-get install.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/kubernetes/test-infra/447#-:-KPxaMS061-TZlnG0N66:b-ambz6j)*

**Comment by apelisse** [comment]
What's the rationale?

---

Review status: 0 of 4 files reviewed at latest revision, 5 unresolved discussions.

---

*[github-stats/fetcher/Dockerfile, line 16 \[r1\]](https://reviewable.io:443/reviews/kubernetes/test-infra/447#-KPx_k0dVK01hQyOPD8t:-KPxborsDP_gjmcgoSWi:b87p2in) ([raw file](https://github.com/kubernetes/test-infra/blob/5eb8377311aa0d8d1fc9ac56d0d3f4e8c2ff09eb/github-stats/fetcher/Dockerfile#L16)):*
<details><summary><i>Previously, spxtr (Joe Finney) wrote…</i></summary>
> The apt-get update needs to be in the same command as the apt-get install.
</details>
Why is that?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/kubernetes/test-infra/447#-:-KPxbrAnYelsrmwsIpkQ:bf29v8i)*

**Comment by spxtr** [comment]
It's just how we usually do things. There are a few advantages but nothing major.

---

Review status: 0 of 4 files reviewed at latest revision, 6 unresolved discussions.

---

*[github-stats/fetcher/deployment.yaml, line 15 \[r1\]](https://reviewable.io:443/reviews/kubernetes/test-infra/447#-KPxfyMiCuLKqhys2lL4:-KPxfyMiCuLKqhys2lL5:b-a0hdwi) ([raw file](https://github.com/kubernetes/test-infra/blob/5eb8377311aa0d8d1fc9ac56d0d3f4e8c2ff09eb/github-stats/fetcher/deployment.yaml#L15)):*
> ```YAML
>       - name: fetcher
>         command:
>         - /fetcher
> ```

You can set this as the entrypoint to the image and just have args.

---

*[github-stats/fetcher/Dockerfile, line 16 \[r1\]](https://reviewable.io:443/reviews/kubernetes/test-infra/447#-KPx_k0dVK01hQyOPD8t:-KPxfV6vIhwOxfc6u6g3:bsje2r8) ([raw file](https://github.com/kubernetes/test-infra/blob/5eb8377311aa0d8d1fc9ac56d0d3f4e8c2ff09eb/github-stats/fetcher/Dockerfile#L16)):*
<details><summary><i>Previously, apelisse (Antoine Pelisse) wrote…</i></summary>
> Why is that?
</details>
https://docs.docker.com/engine/userguide/eng-image/dockerfile_best-practices/#/apt-get

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/kubernetes/test-infra/447#-:-KPxfXxqQ4Pn7lFE-yIT:b-tlx67t)*

**Comment by apelisse** [comment]
What are they? :-)

---

Review status: 0 of 4 files reviewed at latest revision, 6 unresolved discussions.

---

*[github-stats/fetcher/Dockerfile, line 16 \[r1\]](https://reviewable.io:443/reviews/kubernetes/test-infra/447#-KPx_k0dVK01hQyOPD8t:-KPxiC0qJ4F3IahvQEB0:b-89q77x) ([raw file](https://github.com/kubernetes/test-infra/blob/5eb8377311aa0d8d1fc9ac56d0d3f4e8c2ff09eb/github-stats/fetcher/Dockerfile#L16)):*
<details><summary><i>Previously, spxtr (Joe Finney) wrote…</i></summary>
> https://docs.docker.com/engine/userguide/eng-image/dockerfile_best-practices/#/apt-get
</details>
Cool!

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/kubernetes/test-infra/447#-:-KPxiIUXfH3HVXhp-t28:b9xz156)*

**Comment by apelisse** [comment]
Review status: 0 of 4 files reviewed at latest revision, 6 unresolved discussions.

---

*[github-stats/fetcher/deploy.sh, line 20 \[r1\]](https://reviewable.io:443/reviews/kubernetes/test-infra/447#-KPxa0SgcR0cejKSj8nc:-KPxmy8pHTmXLVQqokXF:b-896fix) ([raw file](https://github.com/kubernetes/test-infra/blob/5eb8377311aa0d8d1fc9ac56d0d3f4e8c2ff09eb/github-stats/fetcher/deploy.sh#L20)):*
<details><summary><i>Previously, spxtr (Joe Finney) wrote…</i></summary>
> If you're allowing for an argument to the script then you should have a comment at the top saying as much.
</details>
Done.

---

*[github-stats/fetcher/deploy.sh, line 24 \[r1\]](https://reviewable.io:443/reviews/kubernetes/test-infra/447#-KPxaCCVtO0K6ItMVwH9:-KPxmXJ7dYTRRNumatQ_:b-896fix) ([raw file](https://github.com/kubernetes/test-infra/blob/5eb8377311aa0d8d1fc9ac56d0d3f4e8c2ff09eb/github-stats/fetcher/deploy.sh#L24)):*
<details><summary><i>Previously, spxtr (Joe Finney) wrote…</i></summary>
> This is assuming you're running it from the same directory.
</details>
Done.

---

*[github-stats/fetcher/deployment.yaml, line 15 \[r1\]](https://reviewable.io:443/reviews/kubernetes/test-infra/447#-KPxfyMiCuLKqhys2lL4:-KPxmW7R39s96xwNJuRT:b-896fix) ([raw file](https://github.com/kubernetes/test-infra/blob/5eb8377311aa0d8d1fc9ac56d0d3f4e8c2ff09eb/github-stats/fetcher/deployment.yaml#L15)):*
<details><summary><i>Previously, spxtr (Joe Finney) wrote…</i></summary>
> You can set this as the entrypoint to the image and just have args.
</details>
Done.

---

*[github-stats/fetcher/deployment.yaml, line 16 \[r1\]](https://reviewable.io:443/reviews/kubernetes/test-infra/447#-KPxarGfz-NhYxbHqYYd:-KPxmWSm4yHf8ne2BlI7:b-896fix) ([raw file](https://github.com/kubernetes/test-infra/blob/5eb8377311aa0d8d1fc9ac56d0d3f4e8c2ff09eb/github-stats/fetcher/deployment.yaml#L16)):*
<details><summary><i>Previously, spxtr (Joe Finney) wrote…</i></summary>
> This is a bit scary. If anyone manages to see our command line flag then they see our token. Instead, you should mount the secret into a file and pass the location of that.
</details>
Done.

---

*[github-stats/fetcher/Dockerfile, line 15 \[r1\]](https://reviewable.io:443/reviews/kubernetes/test-infra/447#-KPx_gROqTPbx_mBjxTa:-KPxrYoVaVg05meIznJb:b-896fix) ([raw file](https://github.com/kubernetes/test-infra/blob/5eb8377311aa0d8d1fc9ac56d0d3f4e8c2ff09eb/github-stats/fetcher/Dockerfile#L15)):*
<details><summary><i>Previously, spxtr (Joe Finney) wrote…</i></summary>
> Why debian? Why not go with something lighter such as alpine?
</details>
Done.

---

*[github-stats/fetcher/Dockerfile, line 16 \[r1\]](https://reviewable.io:443/reviews/kubernetes/test-infra/447#-KPx_k0dVK01hQyOPD8t:-KPxmawn3kHXfgkY14re:b-896fix) ([raw file](https://github.com/kubernetes/test-infra/blob/5eb8377311aa0d8d1fc9ac56d0d3f4e8c2ff09eb/github-stats/fetcher/Dockerfile#L16)):*
<details><summary><i>Previously, apelisse (Antoine Pelisse) wrote…</i></summary>
> Cool!
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/kubernetes/test-infra/447)*

**Comment by apelisse** [comment]
*[jenkins/job-configs/kubernetes-jenkins/node-e2e.yaml, line 107 \[r3\]](https://reviewable.io:443/reviews/kubernetes/test-infra/447#-KPyxHpGzskvyDJpO30y:-KPyxHpGzskvyDJpO30z:bipc9su) ([raw file](https://github.com/kubernetes/test-infra/blob/3ce605aa966a670a58a627e66347543c14bc12f5/jenkins/job-configs/kubernetes-jenkins/node-e2e.yaml#L107)):*
> ```YAML
>         - e2e-credentials-binding
>         - timeout:
>             timeout: '{test-timeout}'
> ```

This shouldn't be here. Updating.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/kubernetes/test-infra/447)*

**Comment by apelisse** [comment]
Review status: 0 of 7 files reviewed at latest revision, 7 unresolved discussions.

---

*[github-stats/fetcher/deployment.yaml, line 15 \[r1\]](https://reviewable.io:443/reviews/kubernetes/test-infra/447#-KPxfyMiCuLKqhys2lL4:-KPyx1tU0RYaVR_jVxD6:bwgn9la) ([raw file](https://github.com/kubernetes/test-infra/blob/5eb8377311aa0d8d1fc9ac56d0d3f4e8c2ff09eb/github-stats/fetcher/deployment.yaml#L15)):*
<details><summary><i>Previously, apelisse (Antoine Pelisse) wrote…</i></summary>
> Done.
</details>
Actually not done. It didn't work, and I don't really see the point anyway to be honest :-)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/kubernetes/test-infra/447)*

**Comment by fejta** [comment]
It makes the test-infra repo more consistent and consistency is good.

I expect `make deploy` to build and deploy any image in our repo, since that's how all the other images work:

https://github.com/kubernetes/test-infra/blob/master/queue-health/graph/Makefile

---

Review status: 0 of 6 files reviewed at latest revision, 7 unresolved discussions.

---

*[github-stats/fetcher/deploy.sh, line 30 \[r4\]](https://reviewable.io:443/reviews/kubernetes/test-infra/447#-KPz2PXBEfM2B3nMBQ7j:-KPz2PXBEfM2B3nMBQ7k:bpf8d94) ([raw file](https://github.com/kubernetes/test-infra/blob/c5c095368c8336cf04f52e019e0f9d58f4a1f55f/github-stats/fetcher/deploy.sh#L30)):*
> ```Shell
> docker build -t "${IMAGE}" .
> gcloud docker push "${IMAGE}"
> cat deployment.yaml | env - IMAGE="${IMAGE}" envsubst | kubectl apply -f -
> ```

I would much prefer we not use this magic and code the image version into the production deployment. Use latest for a dev one for easy development.

See the various dev/prod queue-health deployments: https://github.com/kubernetes/test-infra/tree/master/queue-health

---

*[github-stats/fetcher/deployment.yaml, line 15 \[r1\]](https://reviewable.io:443/reviews/kubernetes/test-infra/447#-KPxfyMiCuLKqhys2lL4:-KPz1Z9q3Gh7ibayscKa:bx0x7im) ([raw file](https://github.com/kubernetes/test-infra/blob/5eb8377311aa0d8d1fc9ac56d0d3f4e8c2ff09eb/github-stats/fetcher/deployment.yaml#L15)):*
<details><summary><i>Previously, apelisse (Antoine Pelisse) wrote…</i></summary>
> Actually not done. It didn't work, and I don't really see the point anyway to be honest :-)
</details>
It simplifies the interface. I shouldn't need to both with knowing the entrypoint. The image auth should be able to swap out /fetcher for another app without impacting users of the image.

It looks like we definitely at least want /fetcher in an ENTRYPOINT, I'd probably argue for --stderrthreshold=0 as well

https://docs.docker.com/engine/reference/builder/#/entrypoint

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/kubernetes/test-infra/447#-:-KPz1zo8DrHide0vcMwI:b-6oo9fo)*

**Comment by apelisse** [comment]
Review status: 0 of 6 files reviewed at latest revision, 7 unresolved discussions.

---

*[github-stats/fetcher/deploy.sh, line 30 \[r4\]](https://reviewable.io:443/reviews/kubernetes/test-infra/447#-KPz2PXBEfM2B3nMBQ7j:-KPzi5OPrFZ38Vsj-gAi:b-896fix) ([raw file](https://github.com/kubernetes/test-infra/blob/c5c095368c8336cf04f52e019e0f9d58f4a1f55f/github-stats/fetcher/deploy.sh#L30)):*
<details><summary><i>Previously, fejta (Erick Fejta) wrote…</i></summary>
> I would much prefer we not use this magic and code the image version into the production deployment. Use latest for a dev one for easy development.
> 
> See the various dev/prod queue-health deployments: https://github.com/kubernetes/test-infra/tree/master/queue-health
</details>
Done.

---

*[github-stats/fetcher/deployment.yaml, line 15 \[r1\]](https://reviewable.io:443/reviews/kubernetes/test-infra/447#-KPxfyMiCuLKqhys2lL4:-KPzi6nt1RdOm_b3rfFU:b-2x8mv7) ([raw file](https://github.com/kubernetes/test-infra/blob/5eb8377311aa0d8d1fc9ac56d0d3f4e8c2ff09eb/github-stats/fetcher/deployment.yaml#L15)):*
<details><summary><i>Previously, fejta (Erick Fejta) wrote…</i></summary>
> It simplifies the interface. I shouldn't need to both with knowing the entrypoint. The image auth should be able to swap out /fetcher for another app without impacting users of the image.
> 
> It looks like we definitely at least want /fetcher in an ENTRYPOINT, I'd probably argue for --stderrthreshold=0 as well
> 
> https://docs.docker.com/engine/reference/builder/#/entrypoint
</details>
I would love to see an example of this working with Kubernetes.
If I don't include the /fetcher in the command list in deployment, it complains that it can't execute the first parameter. Which means that this only makes me repeast the name of the command.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/kubernetes/test-infra/447)*

**Comment by apelisse** [comment]
Could you explain in more details how this process works?

If I use a dummy image in the deployment file, it means that whenever I apply the deployment again, I end-up with "ErrImgPull" (can't pull the dummy image), and then I have to reset the image with `kubectl set image ...`.

What do I miss?

---

Review status: 0 of 6 files reviewed at latest revision, 7 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/kubernetes/test-infra/447#-:-KPzjUCivGc5KMaZzUva:b-workku)*

File: github-stats/fetcher/deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+      labels:
+        app: fetcher
+    spec:
+      containers:
+      - name: fetcher
+        args:
+        - --token-file=/etc/secret-volume/token
+        - --stderrthreshold=0
+        - --host=sqlproxy-service
+        image: gcr.io/google-containers/fetcher
```
Comment (fejta) [inline github-stats/fetcher/deployment.yaml]: Please use a specific version. Alternatively name this deployment-dev.yaml to make it clear this is not the production version (and if so please create a production yaml)

**Comment by fejta** [comment]
Please specify the "production" version of the image.

Also is this ready for another review? I may be struggling with the reviewable comments...

---

Review status: 0 of 6 files reviewed at latest revision, 8 unresolved discussions.

---

*[github-stats/fetcher/deployment.yaml, line 15 \[r1\]](https://reviewable.io:443/reviews/kubernetes/test-infra/447#-KPxfyMiCuLKqhys2lL4:-KQ32Cdu-haRLPtHpuIS:btw61kc) ([raw file](https://github.com/kubernetes/test-infra/blob/5eb8377311aa0d8d1fc9ac56d0d3f4e8c2ff09eb/github-stats/fetcher/deployment.yaml#L15)):*
<details><summary><i>Previously, apelisse (Antoine Pelisse) wrote…</i></summary>
> I would love to see an example of this working with Kubernetes.
> If I don't include the /fetcher in the command list in deployment, it complains that it can't execute the first parameter. Which means that this only makes me repeast the name of the command.
</details>
Please switch from command to args: http://kubernetes.io/docs/api-reference/v1/definitions/#_v1_container

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/kubernetes/test-infra/447#-:-KQ7Qr-Gdwk7CoqTOIlz:b-aqo4oc)*

**Comment by fejta** [comment]
Reviewed 2 of 6 files at r3, 2 of 5 files at r5, 3 of 3 files at r6.
Review status: all files reviewed at latest revision, 9 unresolved discussions.

---

*[github-stats/fetcher/Makefile, line 2 \[r6\]](https://reviewable.io:443/reviews/kubernetes/test-infra/447#-KQ7t0Gi84jK0-l4MPvF:-KQ7t0Gi84jK0-l4MPvG:b-ha2s1j) ([raw file](https://github.com/kubernetes/test-infra/blob/6b195af4f1ac0ace18be3ce854b9716edaa3299c/github-stats/fetcher/Makefile#L2)):*
> ```Makefile
> # Please set IMG
> TAG := $(shell date +v%s)
> ```

This will be easier to understand as YYYYmmdd like we do elsewhere in the repo. As a human being I cannot efficiently convert 1472249200 into 8/26/2016 @ 15:07 nor notice that 1462249200 is an ancient image that should probably be updated with some hot-fixes.

---

*[github-stats/mysql/README.md, line 20 \[r6\]](https://reviewable.io:443/reviews/kubernetes/test-infra/447#-KQ7ttKRk_lsSw8qcX-d:-KQ7ttKRk_lsSw8qcX-e:b-ssxysl) ([raw file](https://github.com/kubernetes/test-infra/blob/6b195af4f1ac0ace18be3ce854b9716edaa3299c/github-stats/mysql/README.md#L20)):*
> ```Markdown
> ```
> kubectl create configmap sqlproxy-config --from-literal=instances=${gcp_sql_project}:github-database=tcp:0.0.0.0:3306
> kubectl create configmap certificates --from-file=/etc/ssl/certs/ca-certificates.crt
> ```

Hmm... Is certs file on my local workstation really the best place for me to get these certs? I suspect this might have weird extra ones added by my proxies/corporation?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/kubernetes/test-infra/447)*

**Comment by apelisse** [comment]
Review status: all files reviewed at latest revision, 9 unresolved discussions.

---

*[github-stats/fetcher/deployment.yaml, line 15 \[r1\]](https://reviewable.io:443/reviews/kubernetes/test-infra/447#-KPxfyMiCuLKqhys2lL4:-KQ7yG4t7unqveyQ185P:b-896fix) ([raw file](https://github.com/kubernetes/test-infra/blob/5eb8377311aa0d8d1fc9ac56d0d3f4e8c2ff09eb/github-stats/fetcher/deployment.yaml#L15)):*
<details><summary><i>Previously, fejta (Erick Fejta) wrote…</i></summary>
> Please switch from command to args: http://kubernetes.io/docs/api-reference/v1/definitions/#_v1_container
</details>
Done.

---

*[github-stats/fetcher/deployment.yaml, line 18 \[r5\]](https://reviewable.io:443/reviews/kubernetes/test-infra/447#-KPxPRHWz-7lgk5sG_gg-r5-18:-KQ7yGvn7FOnLjcOoeQa:b-896fix) ([raw file](https://github.com/kubernetes/test-infra/blob/d4bc612a1b4be859d17338a2527ce32c95817d25/github-stats/fetcher/deployment.yaml#L18)):*
<details><summary><i>Previously, fejta (Erick Fejta) wrote…</i></summary>
> Please use a specific version. Alternatively name this deployment-dev.yaml to make it clear this is not the production version (and if so please create a production yaml)
</details>
Done.

---

*[github-stats/fetcher/Makefile, line 2 \[r6\]](https://reviewable.io:443/reviews/kubernetes/test-infra/447#-KQ7t0Gi84jK0-l4MPvF:-KQ7xSWdrzyHpSGD4lHW:bfwd2x8) ([raw file](https://github.com/kubernetes/test-infra/blob/6b195af4f1ac0ace18be3ce854b9716edaa3299c/github-stats/fetcher/Makefile#L2)):*
<details><summary><i>Previously, fejta (Erick Fejta) wrote…</i></summary>
> This will be easier to understand as YYYYmmdd like we do elsewhere in the repo. As a human being I cannot efficiently convert 1472249200 into 8/26/2016 @ 15:07 nor notice that 1462249200 is an ancient image that should probably be updated with some hot-fixes.
</details>
YYYYmmdd doesn't let me have multiple image per day ... I'll update to be add HHMMSS

---

*[github-stats/mysql/README.md, line 20 \[r6\]](https://reviewable.io:443/reviews/kubernetes/test-infra/447#-KQ7ttKRk_lsSw8qcX-d:-KQ7y9YFTQJKCM5XFUwh:bz71qmu) ([raw file](https://github.com/kubernetes/test-infra/blob/6b195af4f1ac0ace18be3ce854b9716edaa3299c/github-stats/mysql/README.md#L20)):*
<details><summary><i>Previously, fejta (Erick Fejta) wrote…</i></summary>
> Hmm... Is certs file on my local workstation really the best place for me to get these certs? I suspect this might have weird extra ones added by my proxies/corporation?
</details>
I'll recommend using this: https://curl.haxx.se/ca/cacert.pem
https://curl.haxx.se/docs/caextract.html

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/kubernetes/test-infra/447)*

**Comment by fejta** [comment]
Reviewed 3 of 3 files at r7.
Review status: all files reviewed at latest revision, 7 unresolved discussions.

---

*[jenkins/job-configs/kubernetes-jenkins/node-e2e.yaml, line 107 \[r3\]](https://reviewable.io:443/reviews/kubernetes/test-infra/447#-KQ86659IJYWqCjL5-_L:-KQ86659IJYWqCjL5-_M:bj0lzzh) ([raw file](https://github.com/kubernetes/test-infra/blob/3ce605aa966a670a58a627e66347543c14bc12f5/jenkins/job-configs/kubernetes-jenkins/node-e2e.yaml#L107)):*
> ```YAML
>         - e2e-credentials-binding
>         - timeout:
>             timeout: '{test-timeout}'
> ```

Is this supposed to be here?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/kubernetes/test-infra/447)*

**Comment by fejta** [comment]
LGTM

But it looks like you pulled in some bad node yaml changes?

---

Review status: all files reviewed at latest revision, 7 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/kubernetes/test-infra/447#-:-KQ86STnTCgYXQZ7Gq4G:b-eb01sy)*

**Comment by apelisse** [comment]
*[jenkins/job-configs/kubernetes-jenkins/node-e2e.yaml, line 107 \[r3\]](https://reviewable.io:443/reviews/kubernetes/test-infra/447#-KQ86659IJYWqCjL5-_L:-KQ8JA42Yl6HBJdoNtvL:b4zuto6) ([raw file](https://github.com/kubernetes/test-infra/blob/3ce605aa966a670a58a627e66347543c14bc12f5/jenkins/job-configs/kubernetes-jenkins/node-e2e.yaml#L107)):*
<details><summary><i>Previously, fejta (Erick Fejta) wrote…</i></summary>
> Is this supposed to be here?
</details>
Indeed ..

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/kubernetes/test-infra/447)*

**Comment by apelisse** [comment]
Something I pushed at some point (and remove right away). It's not part of the current most recent version.

---

Review status: all files reviewed at latest revision, 8 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/kubernetes/test-infra/447#-:-KQ948-luPJ2YzfVtjl7:b-ou2n8k)*

**Comment by spxtr** [comment]
Looks good. I would drop the latest tag.

---

Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

*[github-stats/fetcher/Makefile, line 10 \[r7\]](https://reviewable.io:443/reviews/kubernetes/test-infra/447#-KQMzdX28PTji2c6zaHe:-KQMzdX28PTji2c6zaHf:ba4imrf) ([raw file](https://github.com/kubernetes/test-infra/blob/167a36a03960a45c028077c4c75956896209afe2/github-stats/fetcher/Makefile#L10)):*
> ```Makefile
> 	CGO_ENABLED=0 go build .
> 	docker build -t $(IMG):$(TAG) .
> 	docker tag -f $(IMG):$(TAG) $(IMG):latest
> ```

You don't need to tag it with latest, I think that happens automatically. Also, you should never be using latest ever, so I would just drop this.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/kubernetes/test-infra/447#-:-KQMzlg3KjZDitgpsi7a:b5127kf)*

**Comment by fejta** [comment]
Review status: 5 of 6 files reviewed at latest revision, 3 unresolved discussions.

---

*[github-stats/fetcher/Makefile, line 10 \[r7\]](https://reviewable.io:443/reviews/kubernetes/test-infra/447#-KQMzdX28PTji2c6zaHe:-KQNDAxq5OVZCiZp42CW:b-l5n9k6) ([raw file](https://github.com/kubernetes/test-infra/blob/167a36a03960a45c028077c4c75956896209afe2/github-stats/fetcher/Makefile#L10)):*
<details><summary><i>Previously, spxtr (Joe Finney) wrote…</i></summary>
> You don't need to tag it with latest, I think that happens automatically. Also, you should never be using latest ever, so I would just drop this.
</details>
a) I believe this is necessary because it is built with a specific tag.
b) I find latest useful for development and testing

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/kubernetes/test-infra/447)*

**Comment by spxtr** [comment]
Review status: 5 of 6 files reviewed at latest revision, 3 unresolved discussions.

---

*[github-stats/fetcher/Makefile, line 10 \[r7\]](https://reviewable.io:443/reviews/kubernetes/test-infra/447#-KQMzdX28PTji2c6zaHe:-KQNGEy0vKBTrkQvUe5c:bes1irr) ([raw file](https://github.com/kubernetes/test-infra/blob/167a36a03960a45c028077c4c75956896209afe2/github-stats/fetcher/Makefile#L10)):*
<details><summary><i>Previously, fejta (Erick Fejta) wrote…</i></summary>
> a) I believe this is necessary because it is built with a specific tag.
> b) I find latest useful for development and testing
</details>
a) You are correct.
b) The latest tag doesn't actually mean what you'd like for development, though. I might be wrong about the specifics because they're super confusing, but if you already have a :latest on your machine and do a pull then it might not get anything newer. The general best-practice is to always specifically tag images.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/kubernetes/test-infra/447)*

**Comment by apelisse** [comment]
Review status: 5 of 6 files reviewed at latest revision, 3 unresolved discussions.

---

*[github-stats/fetcher/Makefile, line 10 \[r7\]](https://reviewable.io:443/reviews/kubernetes/test-infra/447#-KQMzdX28PTji2c6zaHe:-KQNL27c-18PsK87ehhC:btapaik) ([raw file](https://github.com/kubernetes/test-infra/blob/167a36a03960a45c028077c4c75956896209afe2/github-stats/fetcher/Makefile#L10)):*
<details><summary><i>Previously, spxtr (Joe Finney) wrote…</i></summary>
> a) You are correct.
> b) The latest tag doesn't actually mean what you'd like for development, though. I might be wrong about the specifics because they're super confusing, but if you already have a :latest on your machine and do a pull then it might not get anything newer. The general best-practice is to always specifically tag images.
</details>
Kubernetes automatically repulls if the tag is latest

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/kubernetes/test-infra/447)*

**Comment by fejta** [comment]
Review status: 5 of 6 files reviewed at latest revision, 3 unresolved discussions.

---

*[github-stats/fetcher/Makefile, line 10 \[r7\]](https://reviewable.io:443/reviews/kubernetes/test-infra/447#-KQMzdX28PTji2c6zaHe:-KQNLazK3NHYvW2oeGaG:bpra6c) ([raw file](https://github.com/kubernetes/test-infra/blob/167a36a03960a45c028077c4c75956896209afe2/github-stats/fetcher/Makefile#L10)):*
<details><summary><i>Previously, apelisse (Antoine Pelisse) wrote…</i></summary>
> Kubernetes automatically repulls if the tag is latest
</details>
Note that the behavior of k8s and docker differ here: http://kubernetes.io/docs/user-guide/images/

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/kubernetes/test-infra/447)*

**Pull request merged**